### PR TITLE
adding call stages

### PR DESCRIPTION
**Pull request by benlower**
updated "phone calls" to be "ultravox + telephony". minor content fix in sdk page.

### CODE REVIEW

File: docs/src/content/docs/guides/stages.mdx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+| property            | change with new stage? |
+|-------------------- | ---------------------- |
+| systemPrompt        | Yes                    |
+| temperature         | Yes                    |
+| voice               | Yes                    |
+| languageHint        | Yes                    |
+| initialMessages     | Yes                    |
+| selectedTools       | Yes                    |
+| firstSpeaker        | Yes                    |
```
Comment (mdepinet) [inline docs/src/content/docs/guides/stages.mdx:48]: firstSpeaker can't change either - it only matters for the first stage after all.

File: docs/src/content/docs/guides/stages.mdx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+| model               | No                     |
+| joinTimeout         | No                     |
+| maxDuration         | No                     |
+| timeExceededMessage | No                     |
+| inactivityMessages  | No                     |
+| medium              | No                     |
+| initiator           | No                     |
+| recordingEnabled    | No                     |
+
+The response body must contain the new values for any properties you want to change in the new stage.
```
Comment (mdepinet) [inline docs/src/content/docs/guides/stages.mdx:58]: Maybe mention that this response body schema is a subset of the request body schema used for starting calls.

File: docs/src/content/docs/guides/stages.mdx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+## Creating and Managing Stages
+
+To implement Call Stages in your Ultravox application, follow these steps:
+
+#### 1. Plan Your Stages
+Determine the different phases of your conversation and what prompts or parameters should change at each stage.
+
+#### 2. Implement a Stage Change Tool
+Create a custom tool that will trigger stage changes when called. A new call stage is created when your tool responds with a header called `X-Ultravox-Response-Type` set to `new-stage` in the response.
```
Comment (mdepinet) [inline docs/src/content/docs/guides/stages.mdx:36]: This also works similarly with client-implemented tools.  The tool implementation just needs to return a new-stage responseType.

File: docs/src/content/docs/guides/stages.mdx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+- Stage 5: Hand-off to loan officer
+
+**Switching Contexts** → Scenarios where the agent needs to navigate different contexts. Examples: customer support escalation, triaging IT issues.
+    
+Let's consider what the potential stages might be for **Customer Support**: 
+- Stage 1: Initial greeting and problem identification
+- Stage 2: Troubleshooting
+- Stage 3: Resolution or escalation (to another stage or to a human support agent)
+
+## Conclusion
```
Comment (mdepinet) [inline docs/src/content/docs/guides/stages.mdx:109]: Probably worth adding a section about how call stages impact other rest endpoints, namely that getting a call or its tools/messages will reflect the call's last stage as you'd expect if you don't use stage changes. For anyone knowingly using stages, the stage-centric APIs are likely to be preferable.

File: docs/src/content/docs/guides/stages.mdx [hunk 5]
```diff
@@ -0,0 +1,111 @@
+---
+title: "Call Stages"
+description: Use Ultravox API to create dynamic, multi-stage conversations with granular system prompts.
```
[the remark this answers is not in the document]
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:3]: good call out. updated this text as well as the intro text at the top of the page.

File: docs/src/content/docs/guides/stages.mdx (hunk 1 above)
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:48]: updated

File: docs/src/content/docs/guides/stages.mdx (hunk 2 above)
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:58]: done

File: docs/src/content/docs/guides/stages.mdx (hunk 3 above)
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:36]: added

File: docs/src/content/docs/guides/stages.mdx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+| initiator           | No                     |
+| recordingEnabled    | No                     |
+
+The response body must contain the new values for any properties you want to change in the new stage.
+
+#### 3. Set Up Stage Transitions
+Prompt the agent to use tool calls to initiate stage changes at appropriate points in the conversation.
+
+#### 4. Configure New Stage Parameters
+When transitioning to a new stage, provide the updated parameters (e.g., system prompt) in the tool call.
```
[the remark this answers is not in the document]
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:64]: yeah good call. moved some of this up and combined others to land with three steps

File: docs/src/content/docs/guides/stages.mdx (hunk 4 above)
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:109]: good call

File: docs/src/content/docs/guides/stages.mdx [hunk 7]
```diff
@@ -86,6 +81,50 @@ export const cstitle = "Updating our systemPrompt and selectedTools"
 New stages inherit all properties from the previous stage. You can selectively overwrite properties as needed when defining a new stage. See the [above table](#2-implement-a-stage-change-tool) for more.
```
[the remark this answers is not in the document]
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:81]: done

File: docs/src/content/docs/guides/stages.mdx [hunk 8]
```diff
@@ -33,35 +37,26 @@ To implement Call Stages in your Ultravox application, follow these steps:
...
-| joinTimeout         | No                     |
-| maxDuration         | No                     |
-| timeExceededMessage | No                     |
-| inactivityMessages  | No                     |
-| medium              | No                     |
-| initiator           | No                     |
-| recordingEnabled    | No                     |
+* Respond with a header called `X-Ultravox-Response-Type` set to `new-stage`. This creates the new stage.
+* Provide the updated parameters (e.g., system prompt, tools, voice) for the new stage in the response body.
+* For [client tools](../sdk#client-tools), a new stage can be started by returning an object that has `responseType="new-stage"`.
```
[the remark this answers is not in the document]
Comment (benlower) [reply docs/src/content/docs/guides/stages.mdx:45]: done

**Pull request merged**